include/libcyphal/transport/udp/sessions_tree.hpp

```diff
@@ -0,0 +1,197 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_UDP_SESSIONS_TREE_HPP
+#define LIBCYPHAL_TRANSPORT_UDP_SESSIONS_TREE_HPP
+
+#include "libcyphal/common/cavl/cavl.hpp"
+#include "libcyphal/transport/errors.hpp"
+#include "libcyphal/transport/types.hpp"
+#include "libcyphal/types.hpp"
+
+#include <cetl/cetl.hpp>
+#include <cetl/pf17/cetlpf.hpp>
+
+#include <cstdint>
+#include <functional>
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace udp
+{
+
+/// Internal implementation details of the UDP transport.
+/// Not supposed to be used directly by the users of the library.
+///
+namespace detail
+{
+
+/// @brief Defines a tree of sessions for the UDP transport.
+///
+/// No Sonar cpp:S4963 b/c we do directly handle resources here.
+///
+template <typename Node>
+class SessionsTree final  // NOSONAR cpp:S4963
```

Comment:
I'm pretty sure this should be `SessionTree` and the file would be `session_tree.hpp`.

Reply:
Actually, should we not promote this as a more general-purpose container? Looking at the implementation I don't see anything session-specific here. We will need a similar container at the presentation layer as well soon...

Reply:
There are things which are quite specific:
1. `<Node>` typename expectation:
  a. having `using NodeRef = typename Node::ReferenceWrapper;`
  b. having `.compareWith` by port id
  c. constructable by single port id argument
3. `ensureNewNodeFor` operates with transport level `AnyError` failure

I propose to do now only `SessionsTree` → `SessionTree` renaming. When we will be at presentation, and there will be similar use case for the tree - then try refactor it somehow to be common.